File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for char in s:
            if char == "(":
                stack.append(")")
            elif char == "[":
                stack.append("]")
            elif char == "{":
                stack.append("}")
            elif len(stack) > 0 and stack[-1] == char:
                stack.pop()
            else:
                return False

        return len(stack) == 0

File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution {
   public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> result(temperatures.size(), 0);
        stack<int> st;

        for (int i = 0; i < temperatures.size(); i++) {
            while (!st.empty() && temperatures[i] > temperatures[st.top()]) {
                int diff = i - st.top();
                result[st.top()] = diff;
                st.pop();
            }
            st.push(i);
        }

        return result;
    }
};
